libfs/fs_test.go

```diff
@@ -597,34 +592,37 @@ func TestFileLocking(t *testing.T) {
 	f, err := fs.Create("a")
 	require.NoError(t, err)
 
-	// We don't have the lock yet so this should fail.
-	err = f.Unlock()
-	require.Error(t, err)
-
 	err = f.Lock()
 	require.NoError(t, err)
 
 	err = f.Unlock()
 	require.NoError(t, err)
 
-	// The lock has been released, but we also haven't made any changes. So the
-	// Unlock should just be a ReleaseLock call without needing to write any
-	// MDs. Since ReleaseLock is idempotent, this should succeed.
+	// The lock has been released, and we haven't made any changes.
+	// This should be a no-op.
 	err = f.Unlock()
 	require.NoError(t, err)
 
 	// Make some more change so next Unlock actually needs to write a MD.
 	_, err = fs.Create("c")
 	require.NoError(t, err)
 
-	// Now we do have some stuff that needs to flush, and we don't have the
-	// lock, so this should fail like the first Unlock.
+	// Now we do have some stuff that needs to flush, but we don't
+	// have the lock, so this should complete without flushing
+	// anything.
 	err = f.Unlock()
-	require.Error(t, err)
+	require.NoError(t, err)
```

Comment:
Now that it doesn't error as long as local state agrees the lock is not held, do you think we should add a case here to make sure an expired lock causes `Unlock` to error?

Reply:
Added a test for that, and one for non-force and force pushes for the runner.